suite-native/device/src/selectors.ts

```diff
@@ -139,9 +140,12 @@ const getTotalFiatBalanceNative = ({
 };
 
 export const selectSelectedDeviceTotalFiatBalance = createMemoizedSelector(
-    [selectDeviceAccounts, selectCurrentFiatRates, selectBaseCurrency],
-    (deviceAccounts, rates, localCurrency) =>
-        getTotalFiatBalanceNative({ deviceAccounts, localCurrency, rates }),
+    [selectDeviceAccounts, selectCurrentFiatRates, selectBaseCurrency, selectHasRunningDiscovery],
+    (deviceAccounts, rates, localCurrency, hasRunningDiscovery) =>
+        // do not return any value before discovery is finished to prevent unnecessary rerenders of portfolio graph.
+        hasRunningDiscovery
```

Comment:
Great catch! Although wouldn’t it be better to observe this value directly in the portfolio graph and render it only when the discovery has finished, rather than modifying the selector? If it’s used elsewhere, it could lead to unexpected behavior.

Reply:
Uh, ok, I see now… it’s more complicated than I thought, as it’s used throughout the Portfolio graph. I thought for a moment we could leverage the `isLoading` prop (which is based on `hasRunningDiscovery` as well) and move the selector one level down, but that probably wouldn’t work that well and would be a bigger hassle.
Good job!